templates-pipelines/windows-efi-installer/manifests/windows-efi-installer.yaml

```diff
@@ -0,0 +1,385 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: windows-efi-installer
+spec:
+  params:
+    - description: Download URL to Windows 11 or server 2022 installation ISO (English United States x64 version is needed). You can follow https://www.microsoft.com/en-us/software-download/windows11 or https://www.microsoft.com/en-us/evalcenter/evaluate-windows-server-2022 to get one.
+      name: winImageDownloadURL
+      type: string
+    - name: instanceTypeName
+      type: string
+      description: Name of VirtualMachineClusterInstancetype object
+      default: u1.large
+    - name: instanceTypeKind
+      type: string
+      description: Kind of VirtualMachineInstancetype object
+      default: VirtualMachineClusterInstancetype
+    - name: preferenceName
+      type: string
+      description: Name of VirtualMachineClusterPreference object
+      default: windows.11
+    - name: virtualMachinePreferenceKind
+      type: string
+      description: Kind of virtualMachinePreference object
+      default: VirtualMachineClusterPreference
+    - default: windows11-autounattend
+      description: Name of the ConfigMap containing the sysprep configuration files (autounattend.xml, etc.). For example windows11-autounattend or windows2022-autounattend. It is possible to provide customize ConfigMaps created by the user too.
+      name: autounattendConfigMapName
+      type: string
+    - default: quay.io/kubevirt/virtio-container-disk:v1.1.1
+      description: Reference to the containerdisk containing the virtio-win drivers ISO.
+      name: virtioContainerDiskName
+      type: string
+    - default: win11
+      description: Name of the base DataVolume which is created. Pre-installed Windows VMs can be created from this DataVolume.
+      name: baseDvName
+      type: string
+    - default: kubevirt-os-images
+      description: Namespace of the base DataVolume which is created.
+      name: baseDvNamespace
+      type: string
+    - default: win11
+      description: Name of Windows ISO datavolume
+      name: isoDVName
+      type: string
+  tasks:
+    - name: import-win-iso
+      params:
+        - name: manifest
+          value: |-
+            apiVersion: cdi.kubevirt.io/v1beta1
+            kind: DataVolume
+            metadata:
+              generateName: $(params.isoDVName)-
+            spec:
+              source:
+                http:
+                  url: $(params.winImageDownloadURL)
+              storage:
+                volumeMode: Filesystem
+                resources:
+                  requests:
+                    storage: 9Gi
+        - name: waitForSuccess
+          value: false
+        - name: allowReplace
+          value: true
+      taskRef:
+{% if use_resolver_in_manifests %}
```

Comment:
If the pipeline itself would be published on the hub, could the templating be avoided?

Reply:
The pipeline has to be published on the hub. The templating is here to allow use case, when cluster is empty and user applies pipeline run with resolver.